src/database/database-sqlite3.cpp

```diff
@@ -606,3 +606,140 @@ void PlayerDatabaseSQLite3::listPlayers(std::vector<std::string> &res)
 
 	sqlite3_reset(m_stmt_player_list);
 }
+
+/*
+ * Auth database
+ */
+
+AuthDatabaseSQLite3::AuthDatabaseSQLite3(const std::string &savedir) :
+		Database_SQLite3(savedir, "auth"), AuthDatabase()
+{
+}
+
+AuthDatabaseSQLite3::~AuthDatabaseSQLite3()
+{
+	FINALIZE_STATEMENT(m_stmt_read)
+	FINALIZE_STATEMENT(m_stmt_write)
+	FINALIZE_STATEMENT(m_stmt_create)
+	FINALIZE_STATEMENT(m_stmt_delete)
+	FINALIZE_STATEMENT(m_stmt_list_names)
+}
+
+void AuthDatabaseSQLite3::createDatabase()
+{
+	assert(m_database); // Pre-condition
+
+	SQLOK(sqlite3_exec(m_database,
+			      "CREATE TABLE IF NOT EXISTS `auth` ("
+			      "`id` INTEGER PRIMARY KEY AUTOINCREMENT,"
+			      "`name` VARCHAR(32),"
+			      "`password` VARCHAR(512),"
+			      "`privileges` VARCHAR(512),"
+			      "last_login INTEGER);",
+			      NULL, NULL, NULL),
+			"Failed to create auth table");
+}
+
+void AuthDatabaseSQLite3::initStatements()
+{
+	PREPARE_STATEMENT(read, "SELECT id, name, password, privileges, last_login FROM "
+				"auth WHERE name = ?");
```

Comment:
SELECT and UPDATE: add `LIMIT 1`. It doesn't have to check the entire database.

Only update: please use `SET` in capitals (trivial)

Reply:
NO LIMIT 1 should NEVER be used to when you have a primary this is a SQL non sense

Reply:
@bendeutsch go ahead for a UNIQUE INDEX on the table.
sauth compat is not required here, it's a core backend. Also we can import data outside sauth and import it without problems (except dup names which is a problem in all cases)